clients/python/girder_client/__init__.py

```diff
@@ -743,12 +757,40 @@ def downloadFolderRecursive(self, folderId, dest):
             })
 
             for item in items:
+                _id = item['_id']
+                self.incoming_metadata[_id] = item
+                if sync and _id in self.local_metadata and \
+                        _compare_items(item, self.local_metadata[_id]):
+                    continue
                 self.downloadItem(item['_id'], dest, name=item['name'])
 
             offset += len(items)
             if len(items) < 50:
                 break
 
+    def saveLocalMetadata(self, dest):
+        """
+        Dumps item metadata collected during a folder download.
+
+        :param dest: The local download destination.
+        """
+
+        with open(os.path.join(dest, ".girder_metadata"), "w") as fh:
+            fh.write(json.dumps(self.incoming_metadata))
+
+    def loadLocalMetadata(self, dest):
+        """
+        Reads item metadata from a local folder.
+
+        :param dest: The local download destination.
+        """
+
+        try:
+            with open(os.path.join(dest, ".girder_metadata"), "r") as fh:
+                self.local_metadata = json.loads(fh.read())
+        except OSError:
+            print("Local metadata does not exists. Falling back to download.")
```

Comment:
I know we have done a bad job of this in the python client, but we are trying to stick to camelCase symbol names moving forward.